EngineLayer/ClassicSearch/ClassicSearchEngine.cs

```diff
@@ -39,9 +40,16 @@ public class ClassicSearchEngine : MetaMorpheusEngine
             {
                 TurnoverLabels = (startLabel, endLabel);
             }
-            Proteins = proteinList;
+
             SearchMode = searchMode;
             SpectralLibrary = spectralLibrary;
+            WriteSpectralLibrary = writeSpectralLibrary;
+
+            // if we're doing a spectral library search, we can skip the reverse protein decoys generated by metamorpheus.
+            // we will generate reverse peptide decoys w/ in the code just below this (and calculate spectral angles for them later).
+            // we have to generate the reverse peptides instead of the usual reverse proteins because we generate decoy spectral
+            // library spectra from their corresponding paired target peptides
+            Proteins = proteinList.Where(p => !p.IsDecoy || spectralLibrary == null).ToList();
```

Comment:
It's a little odd to check if the spectral library is null for every protein.
More efficient code would be:
`Proteins = spectralLibrary == null ? proteinList : proteinList.Where(p=> !p.IsDecoy).ToList();`